tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -77,6 +90,25 @@ type AzureProvider struct {
 	instanceType string
 	// workspace to store all the results.
 	resourceTrackerDir string
+	// requiredRules is the set of SG rules that need to be created or deleted
+	requiredRules map[string]*nsgRuleWrapper
```

Comment:
>While I am fine being specific, do you want to create a slice of *nsgRuleWrapper? just to make extension easier because I don't see any validation specific to any protocol like rdp, winrm etc?

Making it a slice would require looping through it when I am assigning existing `SecurityRule` to the wrapper. So instead I made it a map. Hope that satisfies your request.

Reply:
SGTM